my_pyscf/mcpdft/_lpdft.py

```diff
@@ -0,0 +1,444 @@
+import numpy as np
+from scipy import linalg
+from pyscf.lib import logger
+from pyscf import mcpdft, lib
+from pyscf.mcpdft import _dms
+from mrh.my_pyscf.lassi.lassi import las_symm_tuple, iterate_subspace_blocks
+from mrh.my_pyscf.lassi import op_o1
+
+'''
+This file is taken from pyscf-forge and adopted for the LAS wavefunctions.
+'''
+
+
+def weighted_average_densities(mc):
```

Comment:
Just refer to the pyscf_forge `weighted_average_densities` function, since it does the same thing for now (except that it uses mc.weights, which is better).